dask-gateway/requirements.txt

```diff
@@ -0,0 +1,16 @@
+# These are dependencies for installing the dask-gateway package.
+#
+# NOTE: changes to the dependencies here must also be reflected in
+#       ../dev-environment.yaml
+#
+aiohttp
+# FIXME: click 8.0.4 works, but 8.1.0-8.1.2 has found to cause failures for
+#        currently unknown reasons.
+#
+#        This is tracked in https://github.com/dask/dask-gateway/issues/522.
+#
+click<8.1.0
+dask>=2.2.0
```

Comment:
Do we need to ensure that dask and distributed have matching versions?

Reply:
That shouldn't actually matter, but it should end up happening in practice (dask & distributed's version constraints + pip chosing the most recent releases).

Reply:
This file represents the dependencies of the dask-gateway package, I'm thinking as long as `dask-gateway` doesn't have requirements to match something for its own function we shouldn't constrain it here. I think it becomes up to the person defining the following environments to ensure things match.
- where a dask-gateway client runs
- where the scheduler runs
- where the worker runs

We provide a sample environment (dask-gateway Dockerfile) that represent the scheduler and worker, but doesn't manage an environment where the dask-gateway client runs.

I guess the dask-gateway client should be compatible with the dask-gateway-server as well if there is a breaking change making client/server incompatible with each other.

I presume there is some documentation in the dask ecosystem about the relevance of aligning versions to some degree for certain packages etc? Is it perhaps so that the only additional complexity introduced by dask-gateway is the possebility of a need to align dask-gateway the client with dask-gateway-server?

Reply:
I suppose having the REST endpoints like /v1/ is a reasonable defence against breaking changes.